File: backend/app/schemas/verification.py
"""Public verification schemas."""

from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class VerificationFileResponse(BaseModel):
    id: UUID
    kind: str
    label: str
    sha256: str
    created_at: datetime
    download_url: str


class VerificationSignatoryResponse(BaseModel):
    id: UUID
    name: str
    email: str
    cpf: Optional[str] = None
    role_label: Optional[str] = None
    status: str
    signing_order: int
    auth_method: str
    auth_require_email_otp: bool
    viewed_at: Optional[datetime] = None
    identity_name: Optional[str] = None
    identity_email: Optional[str] = None
    identity_phone: Optional[str] = None
    identity_confirmed_at: Optional[datetime] = None
    otp_sent_at: Optional[datetime] = None
    otp_verified_at: Optional[datetime] = None
    terms_accepted_at: Optional[datetime] = None
    accepted_terms_version: Optional[str] = None
    signed_at: Optional[datetime] = None
    refused_at: Optional[datetime] = None
    signature_mode: Optional[str] = None
    selfie_captured: bool = False
    ip_address_at_sign: Optional[str] = None
    user_agent_at_sign: Optional[str] = None


class VerificationIntegrityResponse(BaseModel):
    configured: bool
    signer_name: Optional[str] = None
    issuer_name: Optional[str] = None
    certificate_serial: Optional[str] = None
    valid_until: Optional[str] = None
    profile: Optional[str] = None


class PublicVerificationResponse(BaseModel):
    document_id: UUID
    document_title: str
    status: str
    source_type: str
    verification_code: str
    verification_url: str
    created_at: datetime
    completed_at: Optional[datetime] = None
    signatories_count: int
    signed_signatories_count: int
    public_data_masked: bool = True
    integrity: VerificationIntegrityResponse
    hashes: list[VerificationFileResponse] = []
    signatories: list[VerificationSignatoryResponse] = []

File: frontend/src/services/verification.ts
import axios from "axios";

const publicApi = axios.create({ baseURL: "/api" });

export interface VerificationFile {
  id: string;
  kind: string;
  label: string;
  sha256: string;
  created_at: string;
  download_url: string;
}

export interface VerificationSignatory {
  id: string;
  name: string;
  email: string;
  cpf: string | null;
  role_label: string | null;
  status: string;
  signing_order: number;
  auth_method: string;
  auth_require_email_otp: boolean;
  viewed_at: string | null;
  identity_name: string | null;
  identity_email: string | null;
  identity_phone: string | null;
  identity_confirmed_at: string | null;
  otp_sent_at: string | null;
  otp_verified_at: string | null;
  terms_accepted_at: string | null;
  accepted_terms_version: string | null;
  signed_at: string | null;
  refused_at: string | null;
  signature_mode: string | null;
  ip_address_at_sign: string | null;
  user_agent_at_sign: string | null;
}

export interface VerificationIntegrity {
  configured: boolean;
  signer_name: string | null;
  issuer_name: string | null;
  certificate_serial: string | null;
  valid_until: string | null;
  profile: string | null;
}

export interface VerificationDocument {
  document_id: string;
  document_title: string;
  status: string;
  source_type: string;
  verification_code: string;
  verification_url: string;
  created_at: string;
  completed_at: string | null;
  signatories_count: number;
  signed_signatories_count: number;
  public_data_masked: boolean;
  integrity: VerificationIntegrity;
  hashes: VerificationFile[];
  signatories: VerificationSignatory[];
}

export async function getVerificationDocument(
  code: string
): Promise<VerificationDocument> {
  const res = await publicApi.get<VerificationDocument>(`/verify/${code}`);
  return res.data;
}
